src/lavinmq/queue/stream_queue_message_store.cr

```diff
@@ -88,6 +106,83 @@ module LavinMQ
         {msg_offset, segment, pos}
       end
 
+      def last_offset_by_consumer_tag(consumer_tag : String)
+        if pos = @consumer_offset_positions[consumer_tag]?
+          tx = @consumer_offsets.to_slice(pos, 8)
+          return IO::ByteFormat::LittleEndian.decode(Int64, tx)
+        end
+      end
+
+      private def restore_consumer_offset_positions : Hash(String, Int64)
+        positions = Hash(String, Int64).new
+        return positions if @consumer_offsets.size.zero?
+
+        loop do
+          ctag = AMQ::Protocol::ShortString.from_io(@consumer_offsets)
+          break if ctag.empty?
+          positions[ctag] = @consumer_offsets.pos
+          @consumer_offsets.skip(8)
+        rescue IO::EOFError
+          break
+        end
+        @consumer_offsets.pos = 0 if @consumer_offsets.pos == 1
+        @consumer_offsets.resize(@consumer_offsets.pos)
+        positions
+      end
+
+      def update_consumer_offset(consumer_tag : String, new_offset : Int64)
+        if pos = @consumer_offset_positions[consumer_tag]?
+          IO::ByteFormat::SystemEndian.encode(new_offset, @consumer_offsets.to_slice(pos, 8, false))
+        else
+          store_consumer_offset(consumer_tag, new_offset)
+        end
+      end
+
+      def store_consumer_offset(consumer_tag : String, new_offset : Int64)
+        expand_consumer_offset_file if consumer_offset_file_full?(consumer_tag)
+        @consumer_offsets.write_bytes AMQ::Protocol::ShortString.new(consumer_tag)
+        @consumer_offset_positions[consumer_tag] = @consumer_offsets.size
+        @consumer_offsets.write_bytes new_offset
+      end
+
+      def consumer_offset_file_full?(consumer_tag)
+        (@consumer_offsets.size + consumer_tag.bytesize + 8) >= @consumer_offsets.capacity
+      end
+
+      def expand_consumer_offset_file
+        pos = @consumer_offsets.size
+        @consumer_offset_capacity += 32_768
+        @consumer_offsets = MFile.new(@consumer_offset_path, @consumer_offset_capacity)
+        @consumer_offsets.resize(pos)
```

Comment:
why was `resize` needed? Shouldn't this be enough? 
```suggestion
        @consumer_offsets = MFile.new(@consumer_offsets.path, @consumer_offsets.capacity + 32 * 1024)
```

Reply:
Write_bytes writes at the end of the file, so resize is needed.